index.js

```diff
@@ -13,21 +15,34 @@ function ZipCodeApi() {
 
 ZipCodeApi.prototype.init = function(options) {
   options = options || {};
+  const clientKey = options.clientKey;
   const apiKey = options.apiKey;
-
-  if (!apiKey) {
-    throw new Error('Must set an API key');
+  const domain = options.domain;
+  if (!clientKey || !apiKey || !domain) {
+    throw new Error('Must set a client key, api key, and domain');
   }
 
+  this.clientKey = clientKey;
   this.apiKey = apiKey;
+  this.domain = domain;
 };
 
 ZipCodeApi.prototype.makeRestUrl = function(suffix) {
-  if (!this.apiKey) {
-    throw new Error('Must set an API key');
+  if (!this.clientKey || !this.apiKey || !this.domain) {
+    throw new Error('Must set a client key, api key, and domain');
   }
 
-  return 'https://www.zipcodeapi.com/rest/' + this.apiKey + suffix;
+  const salt = 'Abcd1234'
+  const currentTime = Math.floor(Date.now() / 1000);
+  const sigPrefix = `${currentTime}-6-${salt}`;
+  const sigBase = `${sigPrefix}-${this.domain.toLowerCase()}-${this.apiKey}`;
+  const hash = sha1(sigBase);
+  const authHash = `${sigPrefix}-${hash}`;
+  const endpoint = `https://www.zipcodeapi.com/rest/${this.clientKey}${suffix}?authHash=${authHash}`;
+
+  console.log({endpoint})
+
+  return endpoint;
 };
 
 ZipCodeApi.prototype.lookupZipCode = function (zipCode) {
```

Comment:
@walliby I noticed when testing, `08057` generates a URL with `/info.json/8057/radians`... which does _not_ work. We might want to handle this as a string somewhere.